rules/frappe_correctness.yml

```diff
@@ -222,3 +222,16 @@ rules:
     If $DOCTYPE is a single doctype then using `frappe.db.get_value` is discouraged for fetching value from single doctypes. frappe.db.get_value for single doctype is not type safe, use `frappe.db.get_single_value` instead.
   languages: [python]
   severity: ERROR
+
+- id: frappe-after-save-controller-hook
+  pattern: |
+    class $DOCTYPE(...):
+      def after_save(self):
+        ...
```

Comment:
```suggestion
    class $DOCTYPE($SOMEBASECLASS, ...):
      ...    
      def after_save(self):
        ...
```


This should fix it:
- method can be anywhere in the class body. 
- class should inherit something. (to ensure we aren't catching non-doctype classes)

Reply:
> method can be anywhere in the class body.

had done the ... thing above after_save initially but noticed semgrep can catch it without that as well

<img width="1059" alt="Screenshot 2022-07-18 at 2 45 32 PM" src="https://user-images.githubusercontent.com/32034600/179480979-aaddf49b-856d-443f-b228-9ff743f3e39c.png">